db.py

```diff
@@ -84,6 +82,15 @@ class Researcher(db.Model):
     user = db.relationship('User', backref=db.backref('user_researcher', uselist=False))
 
 
+class ResearcherSupervisor(db.Model):
+    __tablename__ = 'researcher_supervisor'
+    researcher_id = db.Column(db.Integer, db.ForeignKey('researcher.id'), primary_key=True)
+    supervisor_id = db.Column(db.Integer, db.ForeignKey('user.id'), primary_key=True)
+
+    researcher = db.relationship('Researcher', backref=db.backref('researcher_supervisor', lazy=True))
+    supervisor = db.relationship('User', backref=db.backref('supervisor_researcher', lazy=True))
```

Comment:
Why don't you simply call the backref `researchers` so that it is more intuitive to fetch `User.researchers` to get the supervisor researchers ?